app/exercises/six.py

```diff
@@ -0,0 +1,147 @@
+from exercises.base import BaseTab
+from structures.adjacency_list import DirectedAdjacencyList
+
+import networkx as nx
+import tkinter as tk
+import random
+
+from matplotlib.backends.backend_tkagg import FigureCanvasTkAgg, NavigationToolbar2Tk
+from matplotlib.figure import Figure
+from tkinter import ttk, messagebox
+
+
+class ExerciseSixTab(BaseTab):
+    def __init__(self, master=None, **kw):
+        super().__init__(master=master, **kw)
+
+        self.grid_columnconfigure(0, weight=0)  # menu
+        self.grid_columnconfigure(1, weight=0)  # separator
+        self.grid_columnconfigure(2, weight=4)  # text frame
+        self.grid_columnconfigure(3, weight=0)  # separator
+        self.grid_columnconfigure(4, weight=2)  # canvas
+
+        self.grid_rowconfigure(0, weight=1)
+
+        self.add_menu()
+        self.add_vertical_separator(column=1)
+        self.add_text_frame(row=0, column=2)
+        self.add_vertical_separator(column=3)
+        self.add_canvas(row=0, column=4)
+
+    def add_menu(self):
+        menu_frame = ttk.Frame(self)
+        menu_frame.grid(row=0, column=0, sticky='N', padx=10, pady=10)
+
+        ########################### 1 ###########################
+
+        ttk.Label(menu_frame, text='N').grid(row=0, column=0)
+        self.vertices_entry = ttk.Entry(menu_frame, width=10)
+        self.vertices_entry.grid(row=1, column=0, pady=2, padx=2)
+
+        ttk.Button(menu_frame, text='Generuj losowy graf', width=30, command=self.generate_graph)\
+            .grid(row=2, column=0, columnspan=1, pady=10)
+
+        ttk.Label(menu_frame, text='Wyznacz wartości PageRank:').grid(row=3, column=0)
+
+        ttk.Button(menu_frame, text='Metodą błądzenia przypadkowego', width=30, command=self.random_walk_pagerank)\
+            .grid(row=4, column=0, columnspan=1, pady=3)
+    
+        ttk.Button(menu_frame, text='Metodą iteracyjną', width=30, command=self.iterative_pagerank)\
+            .grid(row=5, column=0, columnspan=1, pady=3)
+
+        ttk.Separator(menu_frame, orient='horizontal')\
+            .grid(row=6, column=0, sticky='EW', pady=15)
+        
+        ########################### 2 ###########################
+
+        ttk.Button(menu_frame, text='Wczytaj plik...', width=30, command=self.load_file)\
+            .grid(row=7, column=0, columnspan=2, pady=3)
+
+        ttk.Separator(menu_frame, orient='horizontal')\
+            .grid(row=8, column=0, columnspan=2, sticky='EW', pady=15)
+
+    def add_canvas(self, row, column):
```

Comment:
Maybe add new parameter to base method e.g. enable_networkx with default value False. It's the easiest way of non repeating this code, but you can figure out something better.